### PR TITLE
Add warnings for lower c++ versions

### PR DESCRIPTION
**Pull request by vikaskurapati**
Compiler warnings for CXX versions lower than the required

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -453,12 +453,18 @@ endif()
 
 # C++ compiler settings
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+  if(CMAKE_CXX_COMPILER_VERSION LESS "9.0")
```
Comment (davschneller) [inline CMakeLists.txt:456]: `VERSION_LESS`

https://cmake.org/cmake/help/latest/command/if.html#version-less

File: CMakeLists.txt [hunk 2]
```diff
@@ -453,12 +453,18 @@ endif()
...
+    message(WARNING "GNU Minimum version 9.0 is required")
+  endif()
   target_compile_options(SeisSol-lib INTERFACE -pedantic $<$<OR:$<COMPILE_LANGUAGE:CXX>,$<COMPILE_LANGUAGE:C>>:-Wall -Wextra -Wno-unused-parameter -Wno-unknown-pragmas>)
 
   # using GCC
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
   # using Intel C++
   # todo remove std?, is ffreestanding needed?
+  if(CMAKE_CXX_COMPILER_VERSION LESS "2021")
```
Comment (davschneller) [inline CMakeLists.txt:465]: `VERSION_LESS`

File: CMakeLists.txt [hunk 3]
```diff
@@ -453,12 +453,18 @@ endif()
 
 # C++ compiler settings
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+  if(CMAKE_CXX_COMPILER_VERSION LESS "9.0")
+    message(WARNING "GNU Minimum version 9.0 is required")
```
Comment (davschneller) [inline CMakeLists.txt:457]: -> minimum (maybe more like: GCC versions below 9.0 are not supported)?

File: CMakeLists.txt [hunk 4]
```diff
@@ -453,12 +453,18 @@ endif()
...
+  endif()
   target_compile_options(SeisSol-lib INTERFACE -pedantic $<$<OR:$<COMPILE_LANGUAGE:CXX>,$<COMPILE_LANGUAGE:C>>:-Wall -Wextra -Wno-unused-parameter -Wno-unknown-pragmas>)
 
   # using GCC
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
   # using Intel C++
   # todo remove std?, is ffreestanding needed?
+  if(CMAKE_CXX_COMPILER_VERSION LESS "2021")
+    message(WARNING "Intel Minimum version 2021 is required")
```
Comment (davschneller) [inline CMakeLists.txt:466]: `Intel compiler versions older than 2021 are not (officially) supported`

**Review by davschneller — COMMENTED** [review]
Some small comments. `VERSION_LESS` is probably the right way to compare here (cf. https://github.com/SeisSol/SeisSol/blob/5f05d19847dd07839c313857699cdba7a4ddcf54/CMakeLists.txt#L19-L24 ).

**Review by Thomas-Ulrich — APPROVED** [review]

**Pull request merged**